src/library/EndpointManagement/EndpointManagementClient.cs

```diff
@@ -454,6 +398,20 @@ private async Task EnsureEndpointManagerRunningAsync(CancellationToken cancellat
                                                 cancellationToken: cancellationToken,
                                                 out string outPut);
 
+                    //  exit code not equal to zero occurs when pkexec doesn't have root access and have to retry with sudo
+                    if (launchExitCode != 0 && _platform.IsLinux && !_environmentVariables.IsCodespaces) {
+                         _log.Info($"pkexec failed with exitCode {launchExitCode}, retrying with sudo");

```

Comment:
@Tatsinnit Yes, I will update the comments but whole idea to try with ```sudo``` if pkexec doesn't work and this would not happen 100% but for some users. Also I will update the logger variable.